include/swift/Runtime/Metadata.h

```diff
@@ -4377,6 +4377,14 @@ void swift_getFieldAt(
 void verifyMangledNameRoundtrip(const Metadata *metadata);
 #endif
 
+#if !SWIFT_OBJC_INTEROP
+
+SWIFT_CC(swift) SWIFT_RUNTIME_STDLIB_SPI // SPI for swift-corelibs-foundation
+ConstTargetMetadataPointer<swift::InProcess, TargetTypeContextDescriptor>
```

Comment:
Maybe this should just return a void *, since (at least for now) the callers treat it as an opaque pointer?

Reply:
I would have it use `const TypeContextDescriptor *` instead of the `Target` template, since this will only ever be used in-process. I don't think there's any harm in having it return the real type here. I also don't think the leading underscore is necessary; IMO this makes sense as a general runtime entry point.

Reply:
I'll make those changes.